Code/Core/FileIO/FileIO.cpp

```diff
@@ -147,12 +165,41 @@
 
 // Copy
 //------------------------------------------------------------------------------
-/*static*/ bool FileIO::FileCopy( const char * srcFileName, const char * dstFileName,
-                              bool allowOverwrite )
+/*static*/ bool FileIO::FileCopy( const char* srcFileName, const char* dstFileName,
+	SymlinkBehavior linkBehavior, bool allowOverwrite )
 {
 #if defined( __WINDOWS__ )
-    DWORD flags = COPY_FILE_COPY_SYMLINK;
-    flags = ( allowOverwrite ? flags : flags | COPY_FILE_FAIL_IF_EXISTS );
+	DWORD flags = ( linkBehavior == FOLLOW_NEITHER ) ? COPY_FILE_COPY_SYMLINK : 0U;
+	flags = ( allowOverwrite ? flags : flags | COPY_FILE_FAIL_IF_EXISTS );
+
+	if ( linkBehavior == FOLLOW_SRC && allowOverwrite )
+	{
+		DWORD dwAttrs = GetFileAttributes( dstFileName );
+		if ( dwAttrs != INVALID_FILE_ATTRIBUTES &&
+			( dwAttrs & FILE_ATTRIBUTE_REPARSE_POINT ) )
+		{
+			HANDLE hFile = CreateFile( dstFileName, 0, FILE_SHARE_READ, 0,
+				OPEN_EXISTING, FILE_FLAG_OPEN_REPARSE_POINT | FILE_FLAG_BACKUP_SEMANTICS, 0 );
+			VERIFY( hFile != INVALID_HANDLE_VALUE );
+			char buf[ 4096 ];
+			DWORD bufSize = 0;
+			bool isSymLink = false;
+			if ( DeviceIoControl( hFile, FSCTL_GET_REPARSE_POINT, NULL, 0, buf, sizeof( buf ), &bufSize, NULL ) )
+			{
+				ULONG* pTag = reinterpret_cast<ULONG*>( buf );
+				isSymLink = ( *pTag == IO_REPARSE_TAG_SYMLINK );
+			}
+			CloseHandle( hFile );
+
+			// CopyFileEx only supports FOLLOW_SRC_AND_DST and FOLLOW_NEITHER
+			// based on the COPY_FILE_COPY_SYMLINK flag.  
+			// (and COPY_FILE_COPY_SYMLINK isn't really FOLLOW_NEITHER)
```

Comment:
The documentation is that the new link will point to the same file as the old link, which leads me to believe that even if you make a relative symlink, COPY_LINK isn't going to produce a link relative to the new path.  But I haven't actually tested that assumption